### DISCUSSION TITLE
How to combine i18n with baseUrl /docs?

### DISCUSSION THREAD

**Discussion by unsphere** [fuma-nama/fumadocs#485] [Q&A] [answered]
I am not able to get fumadocs to work correctly when using [lang] and docs folder together and get the following issues:

- index.mdx becomes the Next button link on the actual index page itself (/en/docs)
- Links (Sidebar,Next/Previous) append lang parameter, e.g.: /en/docs/en or /en/docs/en/test

The language switcher is working correctly. You can reproduce this with the i18n example by adding the docs folder inside of the [lang] folder and changing the baseUrl in source.ts to /docs

Do you have an idea how to set it up correctly?

**Comment by fuma-nama** [comment, marked as answer]
Sorry for late, it should be fixed on the latest release `12.1.2`

**Reply by unsphere** [reply]
Thank you very much! This was not late, this was actually really fast ;)

Does the middleware support hiding the lang prefix if lang equals defaultLanguage? For example defaultLanguage equals "en" then path should be /docs and not /en/docs. Something like [next-i18n-router](https://github.com/i18nexus/next-i18n-router?tab=readme-ov-file#locale-path-prefixing) has prefixDefault: false.

**Reply by fuma-nama** [reply]
Not supported, welcome to open a feature request about it
